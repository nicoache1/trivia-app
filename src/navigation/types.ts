import { ParamListBase } from '@react-navigation/native'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import React from 'react'

import { Routes } from './routes'

export interface ParamList extends ParamListBase {
  [Routes.Welcome]: undefined
  [Routes.ConfigureGame]: undefined
  [Routes.Game]: undefined
  [Routes.Results]: undefined
  [Routes.AnimatedSplash]: undefined
}

export type SceneProps<T extends keyof ParamList> = React.FC<
  NativeStackScreenProps<ParamList, T>
>

export type NavigationProp<T extends keyof ParamList> = NativeStackScreenProps<
  ParamList,
  T
>['navigation']
