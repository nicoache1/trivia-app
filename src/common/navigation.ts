import {
  NavigationContainerRef,
  ParamListBase,
  StackActions,
} from '@react-navigation/native'
import * as React from 'react'

import { ParamList } from '../navigation/types'

export const navigationRef: React.RefObject<NavigationContainerRef<ParamList>> =
  React.createRef()

export function navigate(name: string, params?: ParamListBase[string]) {
  navigationRef.current?.navigate(name, params)
}

export function push(name: string, params?: ParamListBase[string]) {
  navigationRef.current?.dispatch(StackActions.push(name, params))
}

export function popToTop() {
  navigationRef.current?.dispatch(StackActions.popToTop())
}

export function pop() {
  navigationRef.current?.dispatch(StackActions.pop())
}
