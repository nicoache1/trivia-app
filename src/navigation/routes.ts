export enum Routes {
  Game = 'Game',
  ConfigureGame = 'ConfigureGame',
  AnimatedSplash = 'AnimatedSplash',
  Results = 'Results',
  Welcome = 'Welcome',
}
