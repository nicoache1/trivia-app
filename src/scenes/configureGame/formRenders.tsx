import SegmentedControl, {
  SegmentedControlProps,
} from '@react-native-segmented-control/segmented-control'
import { localization } from 'localization'
import React from 'react'
import { ControllerProps } from 'react-hook-form'
import { StyledInput } from 'src/components/StyledInput'

import { ConfigureGameForm } from './types'

export const renderSegmentedControl: <T extends keyof ConfigureGameForm>(
  values: SegmentedControlProps['values'],
  tintColor: SegmentedControlProps['tintColor'],
) => ControllerProps<ConfigureGameForm, T>['render'] =
  (values, tintColor) =>
  ({ field: { onChange, value } }) =>
    (
      <SegmentedControl
        values={values}
        backgroundColor={'rgba(0, 184, 147, 0.8)'}
        tintColor={tintColor}
        selectedIndex={value as number}
        onChange={event => onChange(event.nativeEvent.selectedSegmentIndex)}
      />
    )

export const renderInputControl: ControllerProps<
  ConfigureGameForm,
  'name'
>['render'] = ({
  field: { onChange, onBlur, value },
  formState: { errors },
}) => (
  <StyledInput
    value={value}
    label={localization('playerNameLabel')}
    onChangeText={onChange}
    onBlur={onBlur}
    autoCapitalize={'none'}
    placeholder={localization('playerNamePlaceholder')}
    helperText={localization('playerNameErrorText')}
    error={errors.name !== undefined}
  />
)
