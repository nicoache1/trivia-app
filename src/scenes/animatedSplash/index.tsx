import SplashImage from 'assets/splashShape.png'
import * as SplashScreen from 'expo-splash-screen'
import React, { useLayoutEffect } from 'react'
import { Dimensions, StyleSheet, View } from 'react-native'
import Animated, {
  interpolate,
  runOnJS,
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated'
import { Routes } from 'src/navigation/routes'
import { SceneProps } from 'src/navigation/types'
import { useTheme } from 'src/styles/Theme'

import { styles } from './styles'

const { width, height } = Dimensions.get('window')

export const AnimatedSplash: SceneProps<Routes.AnimatedSplash> = ({
  navigation,
}) => {
  useLayoutEffect(() => {
    navigation.setOptions({ headerShown: false })
  }, [navigation])

  const {
    Theme: { colors },
  } = useTheme()

  const progress = useSharedValue(0)
  const imageAnimation = useSharedValue(0)

  const onImageLoaded = async () => {
    await SplashScreen.hideAsync()
    progress.value = withTiming(1, { duration: 700 }, () => {
      imageAnimation.value = withTiming(1, {}, finished => {
        if (finished) {
          runOnJS(navigation.replace)(Routes.Welcome)
        }
      })
    })
  }

  const animatedStyle = useAnimatedStyle(() => ({
    backgroundColor: colors.SECONDARY,
    transform: [
      { translateY: width / 2 },
      { rotate: `${interpolate(progress.value, [0, 1], [-25, 0])}deg` },
      { translateX: interpolate(progress.value, [0, 1], [-height, 0]) },
      { translateY: -width / 2 },
    ],
    zIndex: 1,
  }))

  const animatedImage = useAnimatedStyle(() => ({
    height: interpolate(imageAnimation.value, [0, 0.5, 1], [70, 80, 0]),
    opacity: interpolate(imageAnimation.value, [0, 1], [1, 0]),
    zIndex: 100,
  }))

  return (
    <View style={styles.container}>
      <Animated.Image
        onLoadEnd={onImageLoaded}
        source={SplashImage}
        style={animatedImage}
        resizeMode={'contain'}
      />
      <Animated.View
        style={[animatedStyle, { ...StyleSheet.absoluteFillObject }]}
      />
    </View>
  )
}
