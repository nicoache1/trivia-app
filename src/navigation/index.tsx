import { createNativeStackNavigator } from '@react-navigation/native-stack'
import React from 'react'

import { AnimatedSplash } from '../scenes/animatedSplash'
import { ConfigureGame } from '../scenes/configureGame'
import { Game } from '../scenes/game'
import { Results } from '../scenes/results'
import { Welcome } from '../scenes/welcome'
import { Routes } from './routes'
import { ParamList } from './types'

const Stack = createNativeStackNavigator<ParamList>()

export const AppNavigator: React.FC<{}> = ({}) => {
  return (
    <Stack.Navigator
      screenOptions={{
        contentStyle: {
          backgroundColor: 'white',
        },
      }}>
      <Stack.Screen name={Routes.AnimatedSplash} component={AnimatedSplash} />
      <Stack.Screen
        name={Routes.Welcome}
        component={Welcome}
        options={{ presentation: 'modal' }}
      />
      <Stack.Screen name={Routes.ConfigureGame} component={ConfigureGame} />
      <Stack.Screen name={Routes.Game} component={Game} />
      <Stack.Screen name={Routes.Results} component={Results} />
    </Stack.Navigator>
  )
}
