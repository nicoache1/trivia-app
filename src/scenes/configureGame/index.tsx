import ConfigureImage from 'assets/configure.png'
import { localization } from 'localization'
import React, { useLayoutEffect } from 'react'
import { Controller, useForm } from 'react-hook-form'
import { Dimensions, Image, View } from 'react-native'
import { Background } from 'src/components/Background'
import { SceneContainer } from 'src/components/SceneContainer'
import { StyledButton } from 'src/components/StyledButton'
import { ButtonMode } from 'src/components/StyledButton/ButtonMode'
import { StyledContainer } from 'src/components/StyledContainer'
import { StyledText } from 'src/components/StyledText'
import { Routes } from 'src/navigation/routes'
import { SceneProps } from 'src/navigation/types'
import { useAppDispatch } from 'src/store'
import { setGameAction } from 'src/store/storage/game'
import { useTheme } from 'src/styles/Theme'
import { AppTypography, ColorPalette } from 'src/styles/types'
import { Game, getDifficulties, getQuestionAmounts } from 'src/types/game'

import { renderInputControl, renderSegmentedControl } from './formRenders'
import { styles } from './styles'
import { ConfigureGameForm } from './types'

const { width, height } = Dimensions.get('window')

export const ConfigureGame: SceneProps<Routes.ConfigureGame> = ({
  navigation,
}) => {
  const {
    Theme: { colors },
  } = useTheme()

  const dispatch = useAppDispatch()

  useLayoutEffect(() => {
    navigation.setOptions({
      headerShown: false,
    })
  }, [navigation])

  const { control, handleSubmit } = useForm<ConfigureGameForm>({
    defaultValues: {
      amount: 0,
      difficulty: 0,
      name: localization('defaultName'),
    },
  })

  const difficulties = getDifficulties()
  const questions = getQuestionAmounts()

  const onSubmit = (data: ConfigureGameForm) => {
    const newGame: Game = {
      difficulty: difficulties[data.difficulty],
      player: data.name,
      questionAmount: questions[data.amount],
      results: [],
    }
    dispatch(setGameAction(newGame))
    navigation.navigate(Routes.Game)
  }

  return (
    <View
      style={[styles.screenContainer, { backgroundColor: colors.SECONDARY }]}>
      <SceneContainer style={styles.scene} barStyle={'light-content'}>
        <View style={styles.headerContainer}>
          <Background />
          <Image
            source={ConfigureImage}
            style={{ height: height / 2.5, width: width * 0.9 }}
            resizeMode={'contain'}
          />
          <View style={styles.textContainer}>
            <StyledText
              color={ColorPalette.TERTIARY}
              typography={AppTypography.H1}>
              {localization('configureTitle1')}
            </StyledText>
            <StyledText
              color={ColorPalette.WHITE}
              typography={AppTypography.H1}>
              {localization('configureTitle2')}
            </StyledText>
          </View>
        </View>
        <View style={styles.formContainer}>
          <Controller
            control={control}
            rules={{
              required: true,
            }}
            render={renderInputControl}
            name={'name'}
          />
          <StyledContainer>
            <View style={styles.subtitle}>
              <StyledText
                color={ColorPalette.WHITE}
                typography={AppTypography.P1_BOLD}>
                {localization('difficultyLabel')}
              </StyledText>
            </View>
            <Controller
              control={control}
              rules={{
                required: true,
              }}
              render={renderSegmentedControl<'difficulty'>(
                difficulties,
                colors.WHITE,
              )}
              name={'difficulty'}
            />
          </StyledContainer>
          <StyledContainer>
            <View style={styles.subtitle}>
              <StyledText
                color={ColorPalette.WHITE}
                typography={AppTypography.P1_BOLD}>
                {localization('questionAmountLabel')}
              </StyledText>
            </View>
            <Controller
              control={control}
              rules={{
                required: true,
              }}
              render={renderSegmentedControl<'amount'>(
                questions.map(item => `${item}`),
                colors.WHITE,
              )}
              name={'amount'}
            />
          </StyledContainer>
        </View>
        <StyledContainer style={styles.buttonContainer}>
          <StyledButton
            mode={ButtonMode.DARK}
            label={localization('getStartedLabel')}
            onPress={handleSubmit(onSubmit)}
          />
        </StyledContainer>
      </SceneContainer>
    </View>
  )
}
