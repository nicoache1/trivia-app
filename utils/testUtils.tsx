import type { PreloadedState } from '@reduxjs/toolkit'
import type { RenderOptions } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import { AppStore, RootState, setupStore } from 'src/store'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export function renderWithProvider(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
  }: ExtendedRenderOptions = {},
) {
  return <Provider store={store}>{ui}</Provider>
}
