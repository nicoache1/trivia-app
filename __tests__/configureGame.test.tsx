import { render } from '@testing-library/react-native'
import React from 'react'
import { Routes } from 'src/navigation/routes'
import { ConfigureGame } from 'src/scenes/configureGame'

import {
  createNavigationPropMock,
  createRoutePropMock,
} from '../__mocks__/navigationMocks'
import { renderWithProvider } from '../utils/testUtils'

test('Expect ConfigureGame to be rendered correctly with default', async () => {
  const mockRoute = createRoutePropMock<Routes.ConfigureGame>(
    Routes.ConfigureGame,
  )

  const mockNavigation = createNavigationPropMock<Routes.ConfigureGame>()

  const enhancedComponent = renderWithProvider(
    <ConfigureGame route={mockRoute} navigation={mockNavigation} />,
  )

  const renderedSnapshot = render(enhancedComponent).toJSON()

  expect(renderedSnapshot).toMatchSnapshot()
})
