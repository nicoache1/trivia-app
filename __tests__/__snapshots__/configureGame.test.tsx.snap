// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect ConfigureGame to be rendered correctly with default 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "rgb(108,92,231)",
      },
    ]
  }
>
  <RNCSafeAreaView
    edges={
      Array [
        "bottom",
      ]
    }
    style={
      Array [
        Object {
          "flexGrow": 1,
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 70,
          }
        }
      >
        <RNSVGSvgView
          align="xMinYMin"
          bbHeight="812"
          bbWidth={750}
          fill="none"
          focusable={false}
          height="812"
          meetOrSlice={1}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 812,
                "width": 750,
              },
            ]
          }
          vbHeight={812}
          vbWidth={375}
          width={750}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M-757.5 612.5L-1210 243.5V948.5H459V264.5L112 673L-335.5 -76L-757.5 612.5Z"
              fill={4279510628}
              fillOpacity={0.3}
              propList={
                Array [
                  "fill",
                  "fillOpacity",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 533.6,
            "width": 675,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessible={true}
          color="TERTIARY"
          style={
            Object {
              "color": "#142664",
              "fontSize": 32,
              "fontStyle": "normal",
              "fontWeight": "700",
            }
          }
          typography="H1"
        >
          configureTitle1
        </Text>
        <Text
          accessible={true}
          color="WHITE"
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 32,
              "fontStyle": "normal",
              "fontWeight": "700",
            }
          }
          typography="H1"
        >
          configureTitle2
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 4,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(231, 231, 231)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#FFFFFF",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "bottom": 0,
                "height": 2,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "transform": Array [
                  Object {
                    "scaleY": 0.5,
                  },
                ],
                "zIndex": 1,
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "minHeight": 64,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "translateX": 3,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 34,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": -16,
                      },
                      Object {
                        "scale": 0.75,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                playerNameLabel
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "opacity": 0,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 34,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": -16,
                      },
                      Object {
                        "scale": 0.75,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
              >
                playerNameLabel
              </Text>
            </View>
            <TextInput
              activeOutlineColor="#FFFFFF"
              autoCapitalize="none"
              editable={true}
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=""
              placeholderTextColor="#FFFFFF"
              selectionColor="#FFFFFF"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                  },
                  Object {
                    "paddingLeft": 12,
                    "paddingRight": 12,
                  },
                  Object {
                    "height": 64,
                  },
                  Object {
                    "paddingBottom": 4,
                    "paddingTop": 24,
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Array [
                    Object {},
                  ],
                ]
              }
              testID="text-input-flat"
              underlineColorAndroid="transparent"
              value="defaultName"
            />
          </View>
        </View>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          onLayout={[Function]}
          style={
            Object {
              "color": "#B00020",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "400",
              "opacity": 0,
              "paddingHorizontal": 12,
              "paddingVertical": 4,
              "textAlign": "left",
              "transform": Array [],
              "writingDirection": "ltr",
            }
          }
        >
          playerNameErrorText
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          }
        }
      >
        <View
          style={
            Object {
              "paddingVertical": 10,
            }
          }
        >
          <Text
            accessible={true}
            color="WHITE"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "700",
              }
            }
            typography="P1_BOLD"
          >
            difficultyLabel
          </Text>
        </View>
        <RNCSegmentedControl
          backgroundColor="rgba(0, 184, 147, 0.8)"
          enabled={true}
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 32,
              },
              undefined,
            ]
          }
          tintColor="#FFFFFF"
          values={
            Array [
              "easy",
              "medium",
              "hard",
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "paddingHorizontal": 20,
            "paddingVertical": 10,
          }
        }
      >
        <View
          style={
            Object {
              "paddingVertical": 10,
            }
          }
        >
          <Text
            accessible={true}
            color="WHITE"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "700",
              }
            }
            typography="P1_BOLD"
          >
            questionAmountLabel
          </Text>
        </View>
        <RNCSegmentedControl
          backgroundColor="rgba(0, 184, 147, 0.8)"
          enabled={true}
          onChange={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 32,
              },
              undefined,
            ]
          }
          tintColor="#FFFFFF"
          values={
            Array [
              "5",
              "10",
              "25",
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 10,
              "height": 56,
              "justifyContent": "center",
              "paddingVertical": 10,
              "width": "100%",
            },
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          animatedStyle={
            Object {
              "value": Object {},
            }
          }
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(0,184,148)",
              "borderRadius": 10,
              "height": 56,
              "justifyContent": "center",
              "paddingVertical": 10,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                },
                Object {
                  "alignItems": "center",
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
            typography="P1_BOLD"
          >
            getStartedLabel
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
